app/jobs/report_job.rb

```diff
@@ -1,15 +1,29 @@
 # frozen_string_literal: true
 
-# Renders a report and saves it to a temporary file.
+# Renders a report and stores it in a given blob.
 class ReportJob < ApplicationJob
-  def perform(report_class, user, params, format)
+  def self.create_blob!
+    # ActiveStorage discourages modifying a blob later but we need a blob
+    # before we know anything about the report file. It enables us to use the
+    # same blob in the controller to read the result.
+    ActiveStorage::Blob.create_before_direct_upload!(
+      filename: "tbd",
+      byte_size: 0,
+      checksum: "0",
+      content_type: "application/octet-stream",
+    ).tap do |blob|
+      ActiveStorage::PurgeJob.set(wait: 1.month).perform_later(blob)
```

Comment:
That's a handy helper!

I can imagine that we'll need to move the expiry duration to a constant or config later, in order to also display it to the user. But that comes later..